common.lic

```diff
@@ -590,6 +590,12 @@ module DRC
 
   def play_song?(settings, song_list, worn = true, skip_clean = false, climbing = false)
     instrument = worn ? settings.worn_instrument : settings.instrument
+    if UserVars.instrument != instrument
+      message("Instrument change detected. Resetting stored song data")
+      UserVars.song = nil
+      UserVars.climbing_song = nil
+    end
+    UserVars.instrument = instrument unless UserVars.instrument == instrument
```

Comment:
Won't this force a reset for every user the first time they play a song after this is merged? Because `UserVars.instrument` will be `nil`, thus seen as a new instrument?

How about
```suggestion
    UserVars.instrument = instrument unless UserVars.instrument
    if UserVars.instrument != instrument
        message("Instrument change detected. Resetting stored song data")
        UserVars.song = nil
        UserVars.climbing_song = nil
        UserVars.instrument = instrument
    end
```

Reply:
Or...

```suggestion
    if !UserVars.instrument # maybe this is UserVars.instrument.nil?
        UserVars.instrument = instrument 
    elsif UserVars.instrument != instrument
        message("Instrument change detected. Resetting stored song data")
        UserVars.song = nil
        UserVars.climbing_song = nil
        UserVars.instrument = instrument
    end
```

Reply:
In principle I like this, however, the reason I am making this change in the first place is that I changed my instruments, went from zills > cowbell, and my song wasn't working, and performance wasn't updating it. I had to manually delete the vars, and re-run performance. 

If we accept nil, someone in my situation would still have to manually delete the var, or change instruments again to reset the song. 

I do accept it might be a good middle-step though, to not force everyone to reset their song when this is pushed... So, all in all, it's a better way forward.

Reply:
How's the current version look?